examples/cpp/LLAMA/dataloader.cc

```diff
@@ -0,0 +1,291 @@
+#include "llama.h"
+#include <random>
+
+using namespace Legion;
+
+DataLoader::DataLoader(FFModel &ff,
+                       LLAMAConfig const *llamaconfig,
+                       ParallelTensor const &input) {
+  Context ctx = ff.config.lg_ctx;
+  Runtime *runtime = ff.config.lg_hlr;
+  num_samples = 347;
+
+  {
+    batch_input = input;
+    int num_dims = input->num_dims;
+
+    ParallelDim dims[num_dims];
+    for (int i = 0; i < num_dims; i++) {
+      dims[i].size = input->dims[i].size;
+      dims[i].degree = 1;
+      dims[i].parallel_idx = -1;
+      dims[i].is_replica_dim = input->dims[i].is_replica_dim;
+      // Assume only the first dim can be the replica dim
+      assert(i == num_dims - 1 || (!dims[i].is_replica_dim));
+    }
+    dims[num_dims - 1].size = num_samples;
+    full_input =
+        ff.create_parallel_tensor_legion_ordering(num_dims, dims, DT_INT64);
+    assert(full_input != nullptr && "full_input is nullptr");
+    ff.map_tensor(full_input, NULL /*parallel_op*/);
+  }
+
+  // Load entire dataset
+  TaskLauncher launcher(CUSTOM_CPU_TASK_ID_1, TaskArgument(NULL, 0));
+  // regions[1]: full_input
+  launcher.add_region_requirement(RegionRequirement(full_input->region,
+                                                    WRITE_ONLY,
+                                                    EXCLUSIVE,
+                                                    full_input->region,
+                                                    MAP_TO_FB_MEMORY));
+  launcher.add_field(0, FID_DATA);
+  runtime->execute_task(ctx, launcher);
+}
+
+void DataLoader::load_entire_dataset(Task const *task,
+                                     std::vector<PhysicalRegion> const &regions,
+                                     Context ctx,
+                                     Runtime *runtime) {
+  assert(regions.size() == 1);
+  assert(task->regions.size() == 1);
+
+  AccessorWO<long, 3> const acc_input(regions[0], FID_DATA);
+  Rect<3> rect_input = runtime->get_index_space_domain(
+      ctx, task->regions[0].region.get_index_space());
+  assert(acc_input.accessor.is_dense_arbitrary(rect_input));
+
+  long *input_ptr = acc_input.ptr(rect_input.lo);
+  std::cout << "load entire dataset" << rect_input.volume();
+
+  // load from file
+  load_from_file(
+      input_ptr,
+      rect_input.volume(),
+      "/home/ubuntu/FlexFlow/examples/cpp/LLAMA/tokens/llama_demo_tokens");
```

Comment:
can you make this path a parameter defined in the `llama.h` header or make it a command line argument?